src/index.ts

```diff
@@ -289,6 +292,26 @@ export class Cat {
     };
   }
 
+  /**
+   * Picks the next item in line from the given list of stimuli.
+   * It grabs the first item from the list, removes it, and then returns it along with the rest of the list.
+   *
+   * @param arr - The list of stimuli to choose from.
+   * @returns {Object} - An object with the next item and the updated list.
+   * @returns {Stimulus} return.nextStimulus - The item that was picked from the list.
+   * @returns {Stimulus[]} return.remainingStimuli - The list of what's left after picking the item.
+   */
+  private selectorFixed(arr: Stimulus[]) {
+    const nextItem = arr.shift();
+    if (nextItem === undefined) {
+      throw new Error('No items left in the array to select.');
+    }
```

Comment:
Is this the behavior we want? Should we throw an error when the user has exhausted all of the available items? We don't do this in any of the other selector methods. I think if we want to do this, we should do it for all of the methods.

Tagging @AnyaWMa and @Emily-ejag

Reply:
I think I added that because it gave me an no-null assertion error.

Do you think changing the next item for this is fine? `const nextItem = arr.shift() ?? null;`